docs/build/guides/transactions/clawbacks.mdx

```diff
@@ -44,354 +44,865 @@ Here we’ll cover the following approaches to clawing back an asset.
+// Helper function to submit transaction and poll for completion using RPC
+function submitAndPollTransaction(transaction, description = "Transaction") {
+  return server.sendTransaction(transaction).then((submitResponse) => {
+    if (submitResponse.status !== "PENDING") {
+      throw new Error(
+        `Transaction submission failed: ${submitResponse.status}`,
+      );
+    }
+
+    console.log(`${description} submitted: ${submitResponse.hash}`);
+
+    return server.pollTransaction(submitResponse.hash).then((finalResponse) => {
+      if (finalResponse.status === "SUCCESS") {
+        console.log(`${description} completed successfully`);
+      } else {
+        console.log(`${description} failed: ${finalResponse.status}`);
+      }
+
+      return {
+        hash: submitResponse.hash,
+        status: finalResponse.status,
+        resultXdr: finalResponse.resultXdr,
+      };
+    });
   });
 }
-```
-
-</CodeExample>
-
-### Example 1: Payments
-
-With the shared setup code out of the way, we can now demonstrate how clawback works for payments. This example will highlight how the asset issuer holds control over their asset regardless of how it gets distributed to the world.
-
-In our scenario, Account A will pay Account B with 1000 tokens of its custom asset; then, B will pay Account C 500 tokens in turn. Finally, A will claw back half of C’s balance, burning 250 tokens forever. Let’s dive into the helper functions:
-
-<CodeExample>
 
-```js
-/// Make a payment to `toAccount` from `fromAccount` for `amount`.
+// Makes payment from `fromAccount` to `toAccount` of `amount`
 function makePayment(toAccount, fromAccount, fromKey, amount) {
-  return server.submitTransaction(
+  console.log(
+    `\nPayment: ${formatAccount(fromAccount.accountId())} → ${formatAccount(
+      toAccount.accountId(),
+    )} (${amount} CLAW)`,
+  );
+  return submitAndPollTransaction(
     buildTx(fromAccount, fromKey, [
       sdk.Operation.payment({
         destination: toAccount.accountId(),
-        asset: ASSET, // defined in preamble
+        asset: ASSET,
+        amount: amount,
+      }),
+    ]),
+    `Payment of ${amount} CLAW`,
+  );
+}
+
+// Creates a claimable balance from `fromAccount` to `toAccount` of `amount`
+function createClaimable(fromAccount, fromKey, toAccount, amount) {
+  console.log(
+    `\nCreating claimable balance: ${formatAccount(
+      fromAccount.accountId(),
+    )} → ${formatAccount(toAccount.accountId())} (${amount} CLAW)`,
+  );
+  return submitAndPollTransaction(
+    buildTx(fromAccount, fromKey, [
+      sdk.Operation.createClaimableBalance({
+        asset: ASSET,
         amount: amount,
+        claimants: [new sdk.Claimant(toAccount.accountId())],
       }),
     ]),
+    `Create Claimable Balance of ${amount} CLAW`,
+  );
+}
+
+// Parse the ClaimableBalanceId from the transaction result XDR
+function getBalanceId(txResponse) {
+  // RPC uses resultXdr, not result_xdr
```

Comment:
This comment elucidates the difference between Horizon and RPC, but since it's all RPC, it might be misleading or throw people off here? Suggest removing it.
```suggestion
```